datafusion/common/src/stats.rs

```diff
@@ -31,6 +31,14 @@ pub enum Precision<T: Debug + Clone + PartialEq + Eq + PartialOrd> {
     Exact(T),
     /// The value is not known exactly, but is likely close to this value
     Inexact(T),
+    /// The value is know to be at most (inclusive) of this value.
+    ///
+    /// The actual value may be smaller, but it is never greater.
+    AtMost(T),
```

Comment:
This is the main change -- adding these variants

Reply:
I wonder if a range semantics would be nicer, i.e.:

```rust
struct Precision {
  lower: Option<T>,
  upper: Option<T>,
}
```

And then you have:

| Big Enum  | Lower     | Upper     |
| --------- | --------- | --------- |
| `Exact`   | `Some(x)` | `Some(x)` |
| `Inexact` | --        | --        |
| `AtMost`  | `None`    | `Some(x)` |
| `AtLeast` | `Some(x)` | `None`    |
| `Absent`  | `None`    | `None`    |
| --        | `Some(x)` | `Some(y)` |